### ISSUE TITLE
Backup for the PostgreSQL database

### ISSUE THREAD

**Issue by mlodic** [honeynet/GreedyBear#16]

**Comment by 0x0elliot** [comment]
Matteo, Would you mind elaborating further on this issue? I want to see if this is a issue I can help fix

**Comment by mlodic** [comment]
It would be cool to have a script that can be launched to perform a backup of the PostgreSQL database in a determined folder in the system that is accessible from the host (not only from the container). In this way, people can then choose where to move those backups files (in a safe place they decide. This is not part of this issue)

Considering that the most important part of this project are the data collected **over time**, it is super important to provide a way to configure automatic backups because if cannot lose the database. (This is extremely different from IntelOwl where the data extracted loses quality over time so we automatically clean up analysis to maintain a clean and little db.)

It could make sense to have a cronjob that runs periodically. It should be configurable:
* the user should choose whether this cron has to be run or not 
* how frequently (every X days)

**Comment by uzaxirr** [comment]
Hey I would like to take up this one.

**Comment by mlodic** [comment]
you already have a PR in progress so no

**Comment by 0x0elliot** [comment]
@uzaxirr if you like, then you can take up this issue right after you're done with your PRs and I wouldn't pick this one up until then. Best of luck! 😄

**Comment by uzaxirr** [comment]
> @uzaxirr if you like, then you can take up this issue right after you're done with your PRs and I wouldn't pick this one up until then. Best of luck! 😄

Haha Thanks, i got assigned somewhat similar work at my internship so i believe i could do it easily

**Comment by Shreyaadgitm** [comment]
Hello sir , I would like to take this issue

**Comment by Shreyaadgitm** [comment]
Hello ,  I am new here . Please tell how to get started , how can I make my first contribution  ?

**Comment by CypherpunkSamurai** [comment]
Hello Sir, I wanted to start contributing to this project and this issue seemed interesting.
I'm currently learning about databases and how to use PostgreSQL with psycopg2, so for gaining a little more experience I would like to work on this issue.

**Things I've Currently Done to Get Acquainted with the project:**
I have currently finished looking through the readme and installation docs. I've tried installing and running the project myself, and have looked through the REST API code, and docker sources.

One of the doubts I'm facing is, I can't figure out how to properly run test the server. The server returns HTTP errors when accessing the rest endpoints. Do we need to provide/simulate a fake TPoT instance for the dev environment?

**Comment by uzaxirr** [comment]
> Hello Sir, I wanted to start contributing to this project and this issue seemed interesting. I'm currently learning about databases and how to use PostgreSQL with psycopg2, so for gaining a little more experience I would like to work on this issue.
> 
> **Things I've Currently Done to Get Acquainted with the project:** I have currently finished looking through the readme and installation docs. I've tried installing and running the project myself, and have looked through the REST API code, and docker sources.
> 
> One of the doubts I'm facing is, I can't figure out how to properly run test the server. The server returns HTTP errors when accessing the rest endpoints. Do we need to provide/simulate a fake TPoT instance for the dev environment?

You need not to simulate TPoT if you just want to visit the endpoint without the feeds data. Can you provide some more information regarding the errors you are facing (Stack Trace or Screenshots)

**Comment by CypherpunkSamurai** [comment]
I'm getting 502 when visiting http://127.0.0.1:80/api/feeds/all/all/recent.json

```shell
$ curl -i  "http://127.0.0.1:80/api/feeds/all/all/recent.json"
```
```HTTP
HTTP/1.1 502 Bad Gateway
Server: nginx
Date: Tue, 15 Mar 2022 15:12:33 GMT
Content-Type: text/html
Content-Length: 150
Connection: keep-alive

<html>
<head><title>502 Bad Gateway</title></head>
<body>
<center><h1>502 Bad Gateway</h1></center>
<hr><center>nginx</center>
</body>
</html>
```

**Logs:**
![image](https://user-images.githubusercontent.com/66906402/158410526-21c64eee-705a-44dc-8be8-260f5f4475a2.png)

**Comment by uzaxirr** [comment]
> I'm getting 502 when visiting http://127.0.0.1:80/api/feeds/all/all/recent.json
> 
> ```shell
> $ curl -i  "http://127.0.0.1:80/api/feeds/all/all/recent.json"
> ```
> 
> ```httpspec
> HTTP/1.1 502 Bad Gateway
> Server: nginx
> Date: Tue, 15 Mar 2022 15:12:33 GMT
> Content-Type: text/html
> Content-Length: 150
> Connection: keep-alive
> 
> <html>
> <head><title>502 Bad Gateway</title></head>
> <body>
> <center><h1>502 Bad Gateway</h1></center>
> <hr><center>nginx</center>
> </body>
> </html>
> ```
> 
> **Logs:** ![image](https://user-images.githubusercontent.com/66906402/158410526-21c64eee-705a-44dc-8be8-260f5f4475a2.png)

Looks like you do not have elasticsearch set up correctly alternatively you can add `DEBUG = True` in your `settings.py` file to run it in development mode this will skip the need of elasticsearch. You can do so by setting it as a env var through `export DEBUG=True`

**Comment by CypherpunkSamurai** [comment]
@uzaxirr It still returns the same errors after enabling debug.

It returns nginx 503 as it previously did.

**Comment by uzaxirr** [comment]
> @uzaxirr It still returns the same errors after enabling debug.

Can you send the terminal's output

PS: It's a nice practice to always provide the terminal's output while asking for help on github

**Comment by CypherpunkSamurai** [comment]
Sure Mr. @uzaxirr.

![image](https://user-images.githubusercontent.com/66906402/158865542-7b558eb5-f3e8-4ae8-8267-75dd0393abf8.png)

I'm receiving the same error even after setting debug as True in the settings.py

**Edit:** Nvm, it worked. :D It appears I forgot to rebuild :P

**Comment by uzaxirr** [comment]
> Sure Mr. @uzaxirr.
> 
> ![image](https://user-images.githubusercontent.com/66906402/158865542-7b558eb5-f3e8-4ae8-8267-75dd0393abf8.png)
> 
> I'm receiving the same error even after setting debug as True in the settings.py

By terminal's output i meant the stack trace and the output that appears in terminal after running docker cmds (The one you sent earlier), but nice to hear that it worked!!

**Comment by CypherpunkSamurai** [comment]
Hello Mr. @mlodic and Mr. @eshaan7, I've added a cronjob and a python module to handle django postgre backups to local file system. I'll be making a draft PR soon, please review the changes / mistakes if any :)

**Edit:**
Please excuse me for using a module, the main reason to use a module rather than raw psycopg2 script is so no to over complicate the codebase. Also by using a module we can exclude ourselves from maintaining the portion, and getting hands dirty with database queries.

**Comment by CypherpunkSamurai** [comment]
Hello Everyone, can someone help me out with the logs?

I've added a few logging statements to the class, but the cron job doesn't seem to show any logs in the terminal

![image](https://user-images.githubusercontent.com/66906402/159552582-6dae8664-2787-4359-b15f-6121691dc9c0.png)

I've made sure the new cron job is added to tasks.py and cellery.py
![image](https://user-images.githubusercontent.com/66906402/159552731-d8f33b6c-fa7c-4590-859f-dce993a98b1d.png)

![image](https://user-images.githubusercontent.com/66906402/159552812-4fd31359-8b79-4696-9b82-41b91fc037fa.png)

**Edit:**
It was just configuration issue, I had forgotten to edit the env file.

**Comment by CypherpunkSamurai** [comment]
Hello Mr. @mlodic and Mr. @eshaan7. I'm was currently trying to debug the celery cron job, while I noticed that other cron jobs are not working as well. 

I edited out the other cron jobs to smaller intervals and waited for logs to pop up, but it didn't. Is there anything else I would needed except debug mode to enable cron jobs?

I also be happy if Mr. @uzaxirr and Mr. @0x0elliot can help me out :)
